src/rail/estimation/algos/deepdisc.py

```diff
@@ -551,12 +629,16 @@ def finalize(self):
         # open each temporary file, write it's contents to the final file.
         is_first = True
         previous_index = 0
+        print(self._temp_file_meta_tuples)
         for meta in self._temp_file_meta_tuples:
             tmp_handle = meta.file_handle
             qp_dstn = tmp_handle.read()
             self._do_chunk_output(qp_dstn, previous_index, previous_index + meta.total_pdfs, is_first)
             previous_index += meta.total_pdfs
             is_first = False
 
+        # finalize the output file
+        self._output_handle.finalize_write()
+
```

Comment:
Did we ever figure out if we the order of lines 641/644 matter?

Reply:
Yes, I do believe that this matters, and here we have them in the correct order. 
`self._output_handle.finalize_write()` will close the output file for writing. Then the call to `PipelineStage.finalize(self)` will rename the output file from "inprogress_<blahblah>.hdf5" to "<blahblah>.hdf5".